OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -464,8 +464,11 @@ public static void SetupReadyWidget(Widget parent, Session.Slot s, Session.Clien
 		public static void AddPlayerFlagAndName(ScrollItemWidget template, Player player)
 		{
 			var flag = template.Get<ImageWidget>("FLAG");
-			flag.GetImageName = () => player.Country.Race;
 			flag.GetImageCollection = () => "flags";
+			if (player.World.RenderPlayer != null && player.World.RenderPlayer.Stances[player] != Stance.Ally)
```

Comment:
Not critical, but I'd change this to `if (player.World.LocalPlayer != null && ...)`. Otherwise the statistics window (F1) shows the random flag when you select a per-player view in observer mode.